src/sinks/nats.rs

```diff
@@ -188,27 +199,31 @@ impl StreamSink<Event> for NatsSink {
 
             let mut bytes = BytesMut::new();
             if self.encoder.encode(event, &mut bytes).is_err() {
-                // Error is logged by `Encoder`.
+                // Error is handled by `Encoder`.
+                finalizers.update_status(EventStatus::Errored);
+                self.acker.ack(1);
                 continue;
-            }
+            };
```

Comment:
Nit: This semicolon is unnecessary now that it's not an statement.

Reply:
Dropped in bec0ec381